get_hg_test.go

```diff
@@ -126,6 +127,11 @@ func TestHgGetter_HgArgumentsNotAllowed(t *testing.T) {
 		t.Log("hg not found, skipping")
 		t.Skip()
 	}
+
+	if runtime.GOOS == "windows" {
+		t.Log("skipping on Windows OS for now")
+		t.Skip()
+	}
```

Comment:
Just to clarify, why are we skipping this test for Windows? 🤔

Reply:
great question (I could've definitely added some sort of comment to clarify this bit) When @nywilken and I were pairing, we kept running issues with getting the windows test to pass with the error below for `hg` -- I'm not entirely sure I understand why, but Wilken may be able to provide more context than I can.

https://github.com/hashicorp/go-getter/actions/runs/3299997899/jobs/5444000536
<img width="862" alt="Screen Shot 2022-10-26 at 11 07 00 AM" src="https://user-images.githubusercontent.com/58451251/198078068-fdb09ae7-43b0-4977-91e2-3022f8b890d3.png">

Reply:
👋 great question. In pairing with @claire-labry I came to believe that these tests were not being previously tested on Windows via CircleCI. So I suggested skipping for now to get the GHA work in place. 

Looking at the v1 code and this code further it appears that only one of the test cases is failing specifically the `"arguments allowed in destination"` test cases, which is not a test case within the v1 code base. 

The test case has a touch command, which I don't think is available on Windows. Maybe that's the issue. An alternative to skipping this whole test on Windows is breaking out the failing case and just skipping that one on Windows for now. I don't have the cycles to dive into it further at this time. So I'm proposing the Windows skip.

Reply:
Totally fair, thank you both for the additional context. It's not preferred, but totally reasonable if you don't have additional cycles. My only worry is that I could imagine this being fairly confusing in a few months time, or for an external contributor.

Perhaps we should capture this in an issue or add a comment to explain why it's only "for now".

Reply:
```suggestion
	if runtime.GOOS == "windows" {
	        // Please refer to https://github.com/hashicorp/go-getter/pull/388/files#r1005819432
	        // for more context why we are temporarily skipping Windows OS.
		t.Log("skipping on Windows OS for now")
		t.Skip()
	}
```

Reply:
Great suggestion-- will add this.